### PR TITLE
Open youtube in brave android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/41958

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Install a version with out the change in the PR
2. disable `Open YouTube links in Brave` option from Settings -> Media
3. Install a new version with the PR change
4. Observe the dialog on the first app open after an upgrade. 
5. Tap `Turn on now`, and observer the redirection to `youtube` page

Adding a recording as the dialog includes an animation. 
https://github.com/user-attachments/assets/60fef9a8-08ba-4bc7-b275-fc0dbec3a243

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/OpenYtInBraveDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        : "moving_ads_light.json";
+        mAnimatedView.setAnimation(animationFile);
+        mAnimatedView.playAnimation();
+    }
+
+    @Override
+    public void onClick(View view) {
+        if (view.getId() == R.id.btn_done) {
+            BravePrefServiceBridge.getInstance().setPlayYTVideoInBrowserEnabled(true);
+            Tab activityTab = BraveRewardsHelper.currentActiveChromeTabbedActivityTab();
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/OpenYtInBraveDialogFragment.java:55]: it's a good idea to move `BraveRewardsHelper.currentActiveChromeTabbedActivityTab` out from `BraveRewardsHelper` to `TabUtils` for example as it seems odd that we use rewards in that place. Plus the function could be refactored a bit as it uses a deprecated call `activity.getTabModelSelector()`

File: browser/ui/android/strings/android_brave_strings.grd [hunk 2]
```diff
@@ -1699,6 +1699,18 @@ Are you sure you want to do this?
       <message name="IDS_BRAVE_STATS_SHARE_TEXT" desc="Text sent when sharing the stats">
         Every day I save data by browsing the Web with Brave.
       </message>
+      <message name="IDS_WATCH_YOUTUBE_NOT_ADS" desc="Open youtube in brave modal title">
+        Watch YouTube, Not Ads.
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:1703]: I believe `not` should not be capitalized here, but defer to @rmcfadden3.

Comment (rmcfadden3) [reply browser/ui/android/strings/android_brave_strings.grd:1703]: Yeah, @mkarolin , you're right. Should be `Watched YouTube, not ads.`

Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:1703]: Thank you @mkarolin @rmcfadden3 updating it in the commit

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by iefremov — APPROVED** [review]

**Pull request merged**